File: Sources/VoxLocalCore/Transcription/WhisperCommandBuilder.swift
import Foundation

/// Pure construction of whisper-cli argument lists (unit-tested).
public enum WhisperCommandBuilder {
    /// - Parameters:
    ///   - outputBase: path *without extension*; whisper-cli writes
    ///     `<outputBase>.json` because of `-oj`.
    public static func arguments(
        modelPath: String,
        audioPath: String,
        language: String,
        threads: Int,
        outputBase: String
    ) -> [String] {
        let clampedThreads = max(1, min(threads, 16))
        return [
            "-m", modelPath,
            "-f", audioPath,
            "-l", language,
            "-t", String(clampedThreads),
            "-oj",              // JSON output for robust parsing
            "-of", outputBase,  // output file base path
            "-np",              // no runtime prints on stdout
        ]
    }
}

File: windows/src/VoxLocal.App/Core/Transcription/WhisperCommandBuilder.cs
namespace VoxLocal.Core.Transcription;

/// <summary>Pure construction of whisper-cli argument lists (unit-tested).</summary>
public static class WhisperCommandBuilder
{
    /// <param name="outputBase">Path *without extension*; whisper-cli writes
    /// &lt;outputBase&gt;.json because of -oj.</param>
    public static IReadOnlyList<string> Arguments(
        string modelPath,
        string audioPath,
        string language,
        int threads,
        string outputBase)
    {
        var clampedThreads = Math.Clamp(threads, 1, 16);
        return new[]
        {
            "-m", modelPath,
            "-f", audioPath,
            "-l", language,
            "-t", clampedThreads.ToString(),
            "-oj",              // JSON output for robust parsing
            "-of", outputBase,  // output file base path
            "-np",              // no runtime prints on stdout
        };
    }
}
